src/dryrun.ts

```diff
@@ -0,0 +1,128 @@
+import AlgodClient from './client/v2/algod/algod';
+import {
+  Application,
+  ApplicationParams,
+  ApplicationStateSchema,
+  DryrunRequest,
+  DryrunSource,
+} from './client/v2/algod/models/types';
+import { SignedTransaction } from './transaction';
+import { TransactionType } from './types/transactions';
+import { encodeAddress } from './encoding/address';
+
+/**
+ * createDryrun takes an Algod Client (from algod.AlgodV2Client) and an array of Signed Transactions
+ * from (transaction.SignedTransaction) and creates a DryrunRequest object with relevant balances
+ * @param client - the AlgodClient to make requests against
+ * @param txns - the array of SignedTransaction to use for generating the DryrunRequest object
+ * @param protocolVersion - the string representing the protocol version to use
+ * @param latestTimestamp - the timestamp
+ * @returns the DryrunRequest object constructed from the SignedTransactions passed
+ */
+export async function createDryrun(
+  client: AlgodClient,
+  txns: SignedTransaction[],
+  protocolVersion?: string,
+  latestTimestamp?: number | bigint,
+  round?: number | bigint,
+  sources?: DryrunSource[]
+): Promise<DryrunRequest> {
```

Comment:
nit: since there are many arguments, some of which are optional, I'd probably suggest using an argument object instead, something like:

```typescript
createDryrun({
  client,
  txns,
  protocolVersion,
  latestTimestamp,
  round,
  sources,
}: {
  client: AlgodClient;
  txns: SignedTransaction[];
  protocolVersion?: string;
  latestTimestamp?: number | bigint;
  round?: number | bigint;
  sources?: DryrunSource[];
}): Promise<DryrunRequest>
```

Reply:
a bit late to the party but I believe none of these can be optional because they are required params for the `DryrunRequest` constructor and they don't have a default value in `createDryrun`